sympy/integrals/integrals.py

```diff
@@ -436,6 +438,8 @@ def doit(self, **hints):
             # See the integrate() docstring for details.
             def try_meijerg(function, xab):
                 ret = None
+                function = function.xreplace(dict([(h, h.rewrite(exp))
+                    for h in function.atoms(HyperbolicFunction)]))
```

Comment:
Exponentials are certainly more simple to express as G-functions than hyperbolics. I wonder if there is any particular reason why this should be so early in integrals.py instead of meijerint.py.; for example, in _meijerint_definite_2 before the _guess_expansion -loop.